### PR TITLE
Cache some web3 queries that shouldn't change

### PR DESCRIPTION
**Pull request by Uxio0**
- Some web3 queries never change and can be cached: `chainId`,
  `getBlockByHash`...
- Use [web3 simple cache middleware](https://web3py.readthedocs.io/en/stable/middleware.html#web3.middleware.construct_simple_cache_middleware)

### CODE REVIEW

File: gnosis/eth/ethereum_client.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
     TimeExhausted,
     TransactionNotFound,
 )
-from web3.middleware import geth_poa_middleware
+from web3.middleware import geth_poa_middleware, simple_cache_middleware
```
Comment (fmrsabino) [inline gnosis/eth/ethereum_client.py:41]: Interesting that they provide this.

It's not clear to me the default behaviour of this middleware though. They do mention in the docs that caching will apply to any call present in `rpc_whitelist`. Do we need to build this `rpc_whitelist`?

https://web3py.readthedocs.io/en/stable/middleware.html#web3.middleware.construct_simple_cache_middleware

Comment (Uxio0) [reply gnosis/eth/ethereum_client.py:41]: No, there's a default list of methods to be cached that shouldn't change: https://github.com/ethereum/web3.py/blob/master/web3/middleware/cache.py#L35

Comment (fmrsabino) [reply gnosis/eth/ethereum_client.py:41]: Ah yes. I see the default argument now! Thanks.

**Review by fmrsabino — APPROVED** [review]
🚀

**Pull request merged**